diff --git a/crypto/CMakeLists.txt b/crypto/CMakeLists.txt
index abe159e..75e398a 100644
--- a/crypto/CMakeLists.txt
+++ b/crypto/CMakeLists.txt
@@ -803,6 +803,7 @@ endif()
 
 if(WIN32)
 	set(CRYPTO_SRC ${CRYPTO_SRC} compat/posix_win.c)
+	set(CRYPTO_SRC ${CRYPTO_SRC} compat/inet_pton.c)
 	set(EXTRA_EXPORT ${EXTRA_EXPORT} gettimeofday)
 	set(EXTRA_EXPORT ${EXTRA_EXPORT} getuid)
 	set(EXTRA_EXPORT ${EXTRA_EXPORT} posix_perror)
