diff --git a/crypto/CMakeLists.txt b/crypto/CMakeLists.txt
index 861ec02..14a6801 100644
--- a/crypto/CMakeLists.txt
+++ b/crypto/CMakeLists.txt
@@ -718,6 +718,7 @@ if(WIN32)
 	set(CRYPTO_SRC ${CRYPTO_SRC} bio/b_win.c)
 	set(CRYPTO_UNEXPORT ${CRYPTO_UNEXPORT} BIO_s_log)
 	set(CRYPTO_SRC ${CRYPTO_SRC} ui/ui_openssl_win.c)
+	set(CRYPTO_SRC ${CRYPTO_SRC} compat/inet_pton.c)
 endif()
 
 if(WIN32)
