diff --git a/crypto/CMakeLists.txt b/crypto/CMakeLists.txt
index d4e0d60..861ec02 100644
--- a/crypto/CMakeLists.txt
+++ b/crypto/CMakeLists.txt
@@ -713,6 +713,7 @@ if(UNIX)
 endif()
 
 if(WIN32)
+	set(CRYPTO_SRC ${CRYPTO_SRC} compat/bcrypt_gen_random.c)
 	set(CRYPTO_SRC ${CRYPTO_SRC} compat/crypto_lock_win.c)
 	set(CRYPTO_SRC ${CRYPTO_SRC} bio/b_win.c)
 	set(CRYPTO_UNEXPORT ${CRYPTO_UNEXPORT} BIO_s_log)
diff --git a/crypto/compat/getentropy_win.c b/crypto/compat/getentropy_win.c
index 64514b3..8ee21ca 100644
--- a/crypto/compat/getentropy_win.c
+++ b/crypto/compat/getentropy_win.c
@@ -41,7 +41,7 @@ getentropy(void *buf, size_t len)
 		return (-1);
 	}
 
-	if (FAILED(BCryptGenRandom(NULL, buf, len, BCRYPT_USE_SYSTEM_PREFERRED_RNG))) {
+	if (FAILED(BCryptGenRandomWrapper(NULL, buf, len, BCRYPT_USE_SYSTEM_PREFERRED_RNG))) {
 		errno = EIO;
 		return (-1);
 	}
