diff --git a/include/spdlog/fmt/xchar.h b/include/spdlog/fmt/xchar.h
index 616d8f5..0c70567 100644
--- a/include/spdlog/fmt/xchar.h
+++ b/include/spdlog/fmt/xchar.h
@@ -8,7 +8,7 @@
 // include bundled or external copy of fmtlib's ostream support
 //
 
-#if !defined(SPDLOG_FMT_EXTERNAL)
+#if 0
 #    ifdef SPDLOG_HEADER_ONLY
 #        ifndef FMT_HEADER_ONLY
 #            define FMT_HEADER_ONLY
