diff --git a/windows/vs2013/config.h b/windows/vs2013/config.h
index 5bd82b0..7c314da 100644
--- a/windows/vs2013/config.h
+++ b/windows/vs2013/config.h
@@ -122,11 +122,17 @@
    makes no measurable difference in performance compared to Vista threads. */
 #define MYTHREAD_WIN95 1
 #else
+
+#if defined(_WIN32_WINNT) and (_WIN32_WINNT <= 0x0501)
+#define MYTHREAD_WIN95 1
+#else
 /* Define to 1 when using Windows Vista compatible threads. This uses features
    that are not available on Windows XP. */
 #define MYTHREAD_VISTA 1
 #endif
 
+#endif
+
 /* Define to 1 to disable debugging code. */
 #define NDEBUG 1
 
diff --git a/windows/vs2017/config.h b/windows/vs2017/config.h
index 25ac18e..2b7c803 100644
--- a/windows/vs2017/config.h
+++ b/windows/vs2017/config.h
@@ -122,11 +122,17 @@
    makes no measurable difference in performance compared to Vista threads. */
 #define MYTHREAD_WIN95 1
 #else
+
+#if defined(_WIN32_WINNT) and (_WIN32_WINNT <= 0x0501)
+#define MYTHREAD_WIN95 1
+#else
 /* Define to 1 when using Windows Vista compatible threads. This uses features
    that are not available on Windows XP. */
 #define MYTHREAD_VISTA 1
 #endif
 
+#endif
+
 /* Define to 1 to disable debugging code. */
 #define NDEBUG 1
 
diff --git a/windows/vs2019/config.h b/windows/vs2019/config.h
index ef921e8..bcd9f91 100644
--- a/windows/vs2019/config.h
+++ b/windows/vs2019/config.h
@@ -122,11 +122,17 @@
    makes no measurable difference in performance compared to Vista threads. */
 #define MYTHREAD_WIN95 1
 #else
+
+#if defined(_WIN32_WINNT) and (_WIN32_WINNT <= 0x0501)
+#define MYTHREAD_WIN95 1
+#else
 /* Define to 1 when using Windows Vista compatible threads. This uses features
    that are not available on Windows XP. */
 #define MYTHREAD_VISTA 1
 #endif
 
+#endif
+
 /* Define to 1 to disable debugging code. */
 #define NDEBUG 1
 
